stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -815,30 +823,45 @@ extension _JSONEncoder {
         }
     }
 
-    fileprivate func box<T : Encodable>(_ value: T) throws -> NSObject {
+    fileprivate func box(_ value: Encodable) throws -> NSObject {
         return try self.box_(value) ?? NSDictionary()
     }
 
+    fileprivate func box(_ dict: [String : Encodable]) throws {
```

Comment:
I think this would be a bit cleaner if it were handled like the other `box(_:)` overloads — this can return an `NSObject` (`NSDictionary` under the hood), and the type can be handled in the `if` like any other case

Reply:
In the latest change, I've tried making it look more like the existing `box(_:)` overloads.